extensions/dacpac/src/wizard/dataTierApplicationWizard.ts

```diff
@@ -313,155 +312,175 @@ export class DataTierApplicationWizard {
-		} catch (e) {
-			additionalProps.exceptionOccurred = 'true';
-		}
+		this.wizard.message = {
+			text: loc.generatingScriptMessage,
+			level: azdata.window.MessageLevel.Information
+		};
 
-		if (!result || !result.success) {
-			vscode.window.showErrorMessage(loc.generateDeployErrorMessage(result?.errorMessage));
-		}
+		const service = await this.getService();
+		const ownerUri = await azdata.connection.getUriForConnection(this.model.server.connectionId);
+
+		const genScriptStartTime = new Date().getTime();
+		const result = await service.generateDeployScript(this.model.filePath, this.model.database, ownerUri, azdata.TaskExecutionMode.script);
 
-		// If result is null which means exception occured, will be adding additional props to the Telemetry
-		if (!result) {
-			additionalProps = { ...additionalProps, ...this.getDacServiceArgsAsProps(service, this.model.database, this.model.filePath, ownerUri) };
-		}
-		additionalProps.isScriptGenerated = result?.success.toString();
-		additionalProps.potentialDataLoss = this.model.potentialDataLoss.toString();
-		additionalMeasurements.deployDacpacFileSizeBytes = await utils.tryGetFileSize(this.model.filePath);
 		additionalMeasurements.totalDurationMs = (new Date().getTime() - genScriptStartTime);
-		// Deploy Dacpac 'generate script' button clicked in DeployPlanPage, Reporting the event selection to the telemetry with fail/sucess status
-		this.sendDacServiceTelemetryEvent(TelemetryViews.DeployDacpac, 'GenerateDeployScriptOperation', additionalProps, additionalMeasurements);
+		additionalMeasurements.deployDacpacFileSizeBytes = await utils.tryGetFileSize(this.model.filePath);
+		additionalProps.potentialDataLoss = this.model.potentialDataLoss.toString();
+
+		if (result?.success) {
+			TelemetryReporter.createActionEvent(TelemetryViews.DataTierApplicationWizard, TelemetryAction.GenerateScript)
+				.withAdditionalProperties(additionalProps)
+				.withAdditionalMeasurements(additionalMeasurements)
+				.send();
+		} else {
+			TelemetryReporter.createErrorEvent(TelemetryViews.DataTierApplicationWizard, TelemetryAction.GenerateScript)
+				.withAdditionalProperties(additionalProps)
+				.withAdditionalMeasurements(additionalMeasurements)
+				.send();
+
+			if (result.errorMessage) {
+				vscode.window.showErrorMessage(loc.generateDeployErrorMessage(result.errorMessage));
+			}
+		}
+
 		return result;
 	}
 
+	public async generateDeployPlan(): Promise<string> {
+		let additionalMeasurements: TelemetryEventMeasures = {};
+
+		const service = await this.getService();
+		const ownerUri = await azdata.connection.getUriForConnection(this.model.server.connectionId);
+
+		const deployPlanStartTime = new Date().getTime();
+		const result = await service.generateDeployPlan(this.model.filePath, this.model.database, ownerUri, azdata.TaskExecutionMode.execute);
+
+		additionalMeasurements.totalDurationMs = (new Date().getTime() - deployPlanStartTime);
+
+		if (result?.success) {
```

Comment:
Could consider consolidating this stuff into a single helper function since most of the code to send the action/error event seems to be the same across all these.